Marlin/configuration_store.cpp

```diff
@@ -215,7 +217,9 @@ typedef struct SettingsDataStruct {
   //
   // HAS_TRINAMIC
   //
-  uint16_t tmc_stepper_current[11];                     // M906 X Y Z X2 Y2 Z2 E0 E1 E2 E3 E4
+  #define TMC_AXES (MAX_EXTRUDERS + 6)
```

Comment:
Something like `TMC_COUNT` would probably be a bit closer but doesn't really matter.

Reply:
The EEPROM needs to save placeholders for all potential axes. It shouldn't resize based on flexible options. `MAX_EXTRUDERS` will only change when Marlin gains support for more extruders.